### PR TITLE
[EFR32] Update gsdk to 4.1

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Change overview
* Update efr32 sdk to 4.1
* Clean silabs submodules to be under "silabs" instead of "efr32_sdk"

#### Testing
* Manual test with efr32 MG12 and MG24
* Automated tests to validate

### CODE REVIEW

File: examples/platform/efr32/ldscripts/efr32mg24.ld [hunk 1]
```diff
@@ -139,44 +144,36 @@ SECTIONS
...
-   * define __STARTUP_CLEAR_BSS_MULTIPLE in startup_ARMCMx.S */
-  /*
   .zero.table :
   {
     . = ALIGN(4);
     __zero_table_start__ = .;
-    LONG (__bss_start__)
-    LONG (__bss_end__ - __bss_start__)
+    /* Add each additional bss section here */
+/*
```
Comment (jmartinez-silabs) [inline examples/platform/efr32/ldscripts/efr32mg24.ld:170]: Again commented code

**Review by jmartinez-silabs — CHANGES_REQUESTED** [review]

File: examples/platform/efr32/ldscripts/efr32mg24.ld [hunk 2]
```diff
@@ -122,11 +114,24 @@ SECTIONS
...
   } > FLASH
-
-
+  /*
+   * SG veneers:
+   * All SG veneers are placed in the special output section .gnu.sgstubs. Its start address
+   * must be set, either with the command line option ‘--section-start’ or in a linker script,
+   * to indicate where to place these veneers in memory.
+   */
+/*
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply examples/platform/efr32/ldscripts/efr32mg24.ld:125]: done

File: examples/platform/efr32/ldscripts/efr32mg24.ld [hunk 3]
```diff
@@ -215,15 +212,56 @@ SECTIONS
...
   } > RAM
 
+  /*
+   * Secondary data section, optional
+   *
+   * Remember to add each additional data section
+   * to the .copy.table above to asure proper
+   * initialization during startup.
+   */
+/*
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply examples/platform/efr32/ldscripts/efr32mg24.ld:222]: done

File: examples/platform/efr32/ldscripts/efr32mg24.ld [hunk 4]
```diff
@@ -45,11 +48,11 @@
...
  ******************************************************************************/
 
- MEMORY
- {
-   FLASH   (rx)  : ORIGIN = 0x8006000, LENGTH = 0x178000
-   RAM     (rwx) : ORIGIN = 0x20000000, LENGTH = 0x40000
- }
+MEMORY
+{
+  FLASH (rx) : ORIGIN = 0x08000000, LENGTH = 0x178000
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply examples/platform/efr32/ldscripts/efr32mg24.ld]: done

File: examples/platform/efr32/ldscripts/efr32mg12.ld [hunk 5]
```diff
@@ -17,10 +17,9 @@
  */
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply examples/platform/efr32/ldscripts/efr32mg12.ld:17]: done

**Review by jmartinez-silabs — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: .github/workflows/examples-efr32.yaml [hunk 6]
```diff
@@ -15,102 +15,121 @@
...
-              if: ${{ !env.ACT }}
-              with:
-                  name: Size,EFR32-Examples,${{ env.GH_EVENT_PR }},${{ env.GH_EVENT_HASH }},${{ env.GH_EVENT_PARENT }},${{ github.event_name }}
-                  path: |
-                      /tmp/bloat_reports/
+      - name: Build some BRD4161A variants
+        timeout-minutes: 50
+        run: |
+          ./scripts/run_in_build_env.sh \
+             "./scripts/build/build_examples.py \
```
Comment (andy31415) [inline .github/workflows/examples-efr32.yaml:94]: Will build_examples.py require to be run only after lfs updates? please make a check/run in the efr32.py logic then. Developers should be able to just compile efr32 builds without worrying about each platform particularities.

**Comment by andy31415** [comment]
Merging as a hotfix with the promise of @mkardous-silabs that updates to dockerfile and build_examples.py will happen quite soon (we talked about reverting the PR if followups are not happening, as a way to ensure they will occur).

File: .github/workflows/examples-efr32.yaml [hunk 7]
```diff
@@ -15,102 +15,121 @@
...
+          with: |
+            token: ${{ github.token }}
+          attempt_limit: 3
+          attempt_delay: 2000
+      - name: Checkout submodules
+        run: scripts/checkout_submodules.py --shallow --platform efr32
 
-        runs-on: ubuntu-latest
-        if: github.actor != 'restyled-io[bot]'
+      - name: Install git-lfs # Temporary measure untill docker container is updated
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply .github/workflows/examples-efr32.yaml:53]: As discussed, this will be addressed in a subsequent PR.

**Comment by andy31415** [comment]
It turns out this breaks cloudbuilds (or generally out of the box builds in vscode image).
May not be easily fixable since we try to avoid running apt installs in images (vscode user has no root rights by default and running stuff as root does not seem right in general out of build scripts)

**Pull request merged**